### ISSUE TITLE
use_spell_check() should run spelling test conditionally

### ISSUE THREAD

**Issue by jennybc** [r-lib/usethis#387]
On my Windows VM, where the Suggested spelling package is not installed, `R CMD check` fails like so:

``` r
* checking tests ...
  Running 'spelling.R'
Warning message:
running command '"C:/PROGRA~1/R/R-34~1.3/bin/x64/R" CMD BATCH --vanilla  "spelling.R" "spelling.Rout"' had status 1 
 ERROR
Running the tests in 'tests/spelling.R' failed.
Complete output:
  > spelling::spell_check_test(vignettes = TRUE, error = FALSE)
  Error in loadNamespace(name) : there is no package called 'spelling'
  Calls: :: ... tryCatch -> tryCatchList -> tryCatchOne -> <Anonymous>
  Execution halted
```

This reveals that `use_spell_check()` should put the spelling test behind `requireNamespace()`.

That is `tests/spelling.R` should look like so:

``` r
if (requireNamespace("spelling", quietly = TRUE)) {
  spelling::spell_check_test(vignettes = TRUE, error = FALSE)
}
```

**Comment by jeroen** [comment]
The template is written here: https://github.com/ropensci/spelling/blob/master/R/spell-check.R#L139-L140. We could add an `if (requireNamespace("spelling",.....`. 

Though usually we assume suggested packages are available for running tests? We also don't condition assume testthat and other packages check suggested packages will be available when running tests?

**Comment by jeroen** [comment]
Yeah that makes sense. I'll try to add this asap.

**Comment by jeroen** [comment]
The template has been fixed in spelling 1.2 (now on CRAN) to test conditionally. For existing packages that had the template generated before, you have to manually update your `tests/spelling.R` file.
